cli/pcluster/config_sanity.py

```diff
@@ -118,6 +118,65 @@ def __check_efs_fs_id(self, ec2, efs, resource_value):  # noqa: C901 FIXME!!!
+            stack_vpc = ec2.describe_subnets(SubnetIds=[resource_value[1]]).get("Subnets")[0].get("VpcId")
+            # Check to see if fs is in the same VPC as the stack
+            if fs.get("VpcId") != stack_vpc:
+                self.__fail(
+                    "VpcId",
+                    "Currently only support using FSx file system that is in the same VPC as the stack. "
+                    "The file system provided is in %s" % fs.get("VpcId"),
+                )
+            # If there is an existing mt in the az, need to check the inbound and outbound rules of the security groups
+            network_interface_ids = fs.get("NetworkInterfaceIds")
+            network_interface_responses = ec2.describe_network_interfaces(
+                NetworkInterfaceIds=network_interface_ids
+            ).get("NetworkInterfaces")
+            network_interfaces = []
+            for response in network_interface_responses:
+                if response.get("VpcId") == stack_vpc:
+                    network_interfaces.append(response)
+            nfs_access = False
+            for network_interface in network_interfaces:
+                in_access = False
+                out_access = False
+                # Get list of security group IDs
+                sg_ids = []
+                for sg_info in network_interface.get("Groups"):
+                    sg_ids.append(sg_info.get("GroupId"))
+                # Check each sg to see if the rules are valid
+                for sg in ec2.describe_security_groups(GroupIds=sg_ids).get("SecurityGroups"):
+                    # Check all inbound rules
+                    in_rules = sg.get("IpPermissions")
+                    for rule in in_rules:
+                        if self.__check_sg_rules_for_port(rule, 988):
+                            in_access = True
+                            break
+                    out_rules = sg.get("IpPermissionsEgress")
+                    for rule in out_rules:
+                        if self.__check_sg_rules_for_port(rule, 988):
+                            out_access = True
+                            break
+                    if in_access and out_access:
+                        nfs_access = True
+                        break
+                if nfs_access:
+                    break
+            if not nfs_access:
+                self.__fail(
+                    "FSXFSId"
+                    "The current security group settings on file system %s does not satisfy "
+                    "mounting requirement. The file system must be associated to a security group that allows "
+                    "inbound and outbound TCP traffic from 0.0.0.0/0 through port 988." % resource_value[0]
+                )
+                sys.exit(1)
+            return True
+        except ClientError as e:
+            self.__fail("FSXFSId", e.response.get("Error").get("Message"))
+
     def validate(self, resource_type, resource_value):  # noqa: C901 FIXME
```

Comment:
I separated the validation logic into a function `__validate_fsx_parameters()` but kept the invocation of that function in the `__validate()` function. This is because in `cfnconfig.py` there's an internal function that checks to see if config sanity is on before calling `validate()` making it hard to have each block in `cfnconfig.py` call a different function.